### PR TITLE
Search: Migrate helper classes to standalone package

### PR DESCRIPTION
**Pull request by jsnmoon**
#### Changes proposed in this Pull Request:
* Moves the Helper and Options classes from the Jetpack plugin in a backward-compatible manner.
* Modifies syntax of existing invocations of the helper classes in the Jetpack plugin.

#### Jetpack product discussion
Part of the standalone Search package/plugin project.

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
* Apply this change to your Jetpack site.
* Ensure that the following features continue working as expected:
  * [x] Classic Search
  * [x] Instant Search
  * [x] Jetpack Search section of the Customizer 
  * [x] Customberg

### CODE REVIEW

**Review by robfelty — COMMENTED** [review]
I think it looks okay to me, but I am just starting to get back into this stuff, and I don't know as much about the jetpack plugin side of things. Do we have unit-tests for these? Do those unit-tests run on WP.com, or another way?

**Review by kangzj — DISMISSED** [review]
The changes look reasonable to me and it works well. Good job 👍

File: tools/phpcs-excludelist.json [hunk 1]
```diff
@@ -18,6 +18,9 @@
 	"projects/packages/analyzer/src/class-differences.php",
 	"projects/packages/analyzer/src/class-invocations.php",
 	"projects/packages/analyzer/src/class-utils.php",
+	"projects/packages/search/src/class-helper.php",
+	"projects/packages/search/src/wpes/query-builder.php",
+	"projects/packages/search/src/wpes/query-parser.php",
```
[the remark this answers is not in the document]
Comment (jsnmoon) [reply tools/phpcs-excludelist.json:23]: Ah, that's a bit of a bother. I'll work on addressing some of the linting errors.

Interestingly, I don't see any required linting checks failing for the PR. Does the aforementioned PHPCS check run outside of feature branches?

**Review by jeherve — DISMISSED** [review]
This seems to test well! 🚢

**Review by jeherve — APPROVED** [review]

**Comment by jsnmoon** [comment]
We will sync changes to WPCOM once we have completed the classic/instant search migration.

**Pull request merged**